File: frontend/src/app/shared/service/admin/homework/HomeWorkEtudiantAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {HomeWorkEtudiantDto} from 'src/app/shared/model/homework/HomeWorkEtudiant.model';
import {HomeWorkEtudiantCriteria} from 'src/app/shared/criteria/homework/HomeWorkEtudiantCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class HomeWorkEtudiantAdminService extends AbstractService<HomeWorkEtudiantDto, HomeWorkEtudiantCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): HomeWorkEtudiantDto {
        return new HomeWorkEtudiantDto();
    }

    public constrcutCriteria(): HomeWorkEtudiantCriteria {
        return new HomeWorkEtudiantCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/homeWorkEtudiant/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/homework/HomeWorkEtudiantAdminService.java
package ma.zs.alc.service.facade.admin.homework;

import java.util.List;
import ma.zs.alc.bean.core.homework.HomeWorkEtudiant;
import ma.zs.alc.dao.criteria.core.homework.HomeWorkEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface HomeWorkEtudiantAdminService extends  IService<HomeWorkEtudiant,HomeWorkEtudiantCriteria>  {

    List<HomeWorkEtudiant> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);
    List<HomeWorkEtudiant> findByHomeWorkId(Long id);
    int deleteByHomeWorkId(Long id);
    long countByHomeWorkId(Long id);




}
